tests/CppUTestContainer.spec.ts

```diff
@@ -18,31 +18,114 @@ describe("CppUTestContainer should", () => {
   let mockResultParser: ResultParser;
 
   beforeEach(() => {
-    mockRunner = createMockRunner("Exec1", "Group1.Test1 Group2.Test2");
+    mockRunner = createMockRunner("Exec1", TestLocationFetchMode.Disabled, "Group1.Test1 Group2.Test2", false);
     mockSetting = mock<SettingsProvider>();
     mockAdapter = mock<VscodeAdapter>();
     mockResultParser = mock<ResultParser>();
     when(mockSetting.GetTestPath()).thenReturn("/test/myPath");
     when(mockSetting.GetTestRunners()).thenReturn(["/test/myPath/Exec1"]);
+    when(mockSetting.TestLocationFetchMode).thenReturn(TestLocationFetchMode.Disabled);
     when(mockResultParser.GetResult(anything())).thenReturn(new TestResult(TestState.Passed, ""));
   })
 
-  it("load all tests from all testrunners", async () => {
-    const mockRunner1 = createMockRunner("Exec1", "Group1.Test1 Group2.Test2");
-    const mockRunner2 = createMockRunner("Exec2", "Group4.Test1 Group5.Test2 Group5.Test42");
+  it("load all tests from all testrunners without location info", async () => {
+    const mockRunner1 = createMockRunner("Exec1", TestLocationFetchMode.Disabled, "Group1.Test1 Group2.Test2", false);
+    const mockRunner2 = createMockRunner("Exec2", TestLocationFetchMode.Disabled, "Group4.Test1 Group5.Test2 Group5.Test42", false);
+    const mockSetting = mock<SettingsProvider>();
+    when(mockSetting.TestLocationFetchMode).thenReturn(TestLocationFetchMode.Disabled);
 
     const container = new CppUTestContainer(instance(mockSetting), instance(mockAdapter), instance(mockResultParser));
 
     const allTests = await container.LoadTests([instance(mockRunner1), instance(mockRunner2)]);
     expect(allTests).to.be.lengthOf(2);
+
     expect(allTests[0].label).to.be.eq("Exec1");
+    expect(allTests[0].children).to.be.lengthOf(2);
```

Comment:
Wow thanks for also writing tests. But could you please split this one? It seems rather large. Other than that: Splendid work!

Reply:
Honestly, I don't see an obvious way to split this test properly.

I've just expanded the test that you already had written, but made it more exhaustive to ensure that all the objects corresponding to the input are properly created and in the right order. Conceptually it's a single test, since it tests the outcome of a single input.

Maybe I could divide the test into a bit more atomic tests like:
- One test that checks that two cpputests from 2 different runners are loaded in the right order.
- One test that tests that 2 cpputests in a single runner from 2 different groups are loaded in the right order.
- One test that tests that 2 cpputests in a single runner from the same group are loaded in the right order.

But in this case we wouldn't be testing strange real-life behaviors like when multiple cpputests are loaded from different runners and groups and the load order is messed up. To test that behaviors we would then have to add more tests which would have to involve more loaded cpputests in different combinations, and at the end we would end-up with tests with more or less the same size than this one.

I can add if you want these more atomic tests to pinpoint issues if the tests break, but anyway when I write tests I always like to have an integration-like test like this one that tests several behaviors together, and the input to this one is small but large enough to check all ordering complexities.

Reply:
Fair enough, you are right it is acutally an integration test and it also is quite holistic in what it does. Thanks